modules/python/test/test_norm.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+
+from itertools import product
+from functools import reduce
+
+import cv2 as cv
+import numpy as np
+
+from tests_common import NewOpenCVTests
+
+
+def norm_inf(x, y=None):
+    def norm(vec):
+        return np.linalg.norm(vec, np.inf)
+
+    return norm(x) if y is None else norm(x-y)
+
+
+def norm_l1(x, y=None):
+    def norm(vec):
+        return np.linalg.norm(vec, 1)
+
+    return norm(x) if y is None else norm(x-y)
+
+
+def norm_l2(x, y=None):
+    def norm(vec):
+        return np.linalg.norm(vec)
+
+    return norm(x) if y is None else norm(x-y)
+
+
+def norm_l2sqr(x, y=None):
+    def norm(vec):
+        return np.square(vec).sum()
+
+    return norm(x) if y is None else norm(x-y)
+
+
+def norm_hamming(x, y=None):
+    def norm(vec):
+        return sum(bin(i).count('1') for i in vec)
+
+    return norm(x) if y is None else norm(np.bitwise_xor(x, y))
+
+
+def norm_hamming2(x, y=None):
+    def norm(vec):
+        def element_norm(element):
+            binary_str = bin(element).split('b')[-1]
+            if len(binary_str) % 2 == 1:
+                binary_str = '0' + binary_str
+            gen = filter(lambda p: p != '00',
+                         (binary_str[i:i+2]
+                          for i in range(0, len(binary_str), 2)))
+            return sum(1 for _ in gen)
+
+        return sum(element_norm(element) for element in vec)
+
+    return norm(x) if y is None else norm(np.bitwise_xor(x, y))
+
+
+norm_type_under_test = {
+    cv.NORM_INF: norm_inf,
+    cv.NORM_L1: norm_l1,
+    cv.NORM_L2: norm_l2,
+    cv.NORM_L2SQR: norm_l2sqr,
+    cv.NORM_HAMMING: norm_hamming,
+    cv.NORM_HAMMING2: norm_hamming2
+}
+
+norm_name = {
+    cv.NORM_INF: 'inf',
+    cv.NORM_L1: 'L1',
+    cv.NORM_L2: 'L2',
+    cv.NORM_L2SQR: 'L2SQR',
+    cv.NORM_HAMMING: 'Hamming',
+    cv.NORM_HAMMING2: 'Hamming2'
+}
+
+
+def get_element_types(norm_type):
+    if norm_type in (cv.NORM_HAMMING, cv.NORM_HAMMING2):
+        return (np.uint8,)
+    else:
+        return (np.int16, np.int32, np.float32, np.float64)
+
+
+def generate_vector(size, dtype):
+    if np.issubdtype(dtype, np.integer):
+        return np.random.randint(0, 100, size).astype(dtype)
```

Comment:
We need reproducible tests, so we need reproducible tests input.

`np.random.randint` generates different numbers on each test launch.

Fix `np.random` seed or use `cv.randu` / `cv.randn`